### ISSUE TITLE
Plugin folder name should match plugin id in manifest.json

### ISSUE THREAD

**Issue by SashaKryzh** [obsidianmd/obsidian-developer-docs#153]
For Obsidian to work correctly, the folder name in `.obsidian/plugins` must match the `id` in `manifest.json`. Without this, methods such as `onExternalSettingsChange` won't be called. However, this requirement isn't mentioned anywhere in the Obsidian documentation (neither on the [Manifest](https://docs.obsidian.md/Reference/Manifest) page nor the [onExternalSettingsChange](https://docs.obsidian.md/Reference/TypeScript+API/Plugin/onExternalSettingsChange) page).

Why this is a problem:

- Many people name their GitHub repositories with the word "obsidian" for organizational purposes, but Obsidian [discourages](https://docs.obsidian.md/Plugins/Releasing/Submit+your+plugin#Step+3+Submit+your+plugin+for+review) including "obsidian" in the `id`. So, if such a repository is cloned, `onExternalSettingsChange` won't work.
- The [Build a plugin](https://docs.obsidian.md/Plugins/Getting+started/Build+a+plugin) guide doesn't include a "rename folder" step, so new developers like me struggle to understand why methods such as `onExternalSettingsChange` don't work ([Discord question](https://discord.com/channels/686053708261228577/840286264964022302/1295352808282918997)).

Solutions:

1. Update the documentation to include a "rename folder" step in the Build a Plugin guide and add this information to the Manifest and `onExternalSettingsChange` pages.
2. Alternatively, changes could be made to Obsidian's code to address this issue, but that seems more complex and unnecessary.

**Comment by SashaKryzh** [comment]
Should I create a PR to update the docs?

**Comment by sigrunixia** [comment]
> Should I create a PR to update the docs?

It would be helpful if you could at least get the foundations started. Joethei just go back from vacation and is catching up on plugin reviews, and my focus is on the main help docs for this week at least. :)

**Comment by SashaKryzh** [comment]
> > Should I create a PR to update the docs?
> 
> It would be helpful if you could at least get the foundations started. Joethei just go back from vacation and is catching up on plugin reviews, and my focus is on the main help docs for this week at least. :)

I created a PR that addresses "Build a plugin" and "Manifest" pages. I think it would be good to also add this info to `onExternalSettingsChange` documentation.

**Comment by sigrunixia** [comment]
> > > Should I create a PR to update the docs?
> > 
> > 
> > It would be helpful if you could at least get the foundations started. Joethei just go back from vacation and is catching up on plugin reviews, and my focus is on the main help docs for this week at least. :)
> 
> I created a PR that addresses "Build a plugin" and "Manifest" pages. I think it would be good to also add this info to `onExternalSettingsChange` documentation.

To confirm, this is roughly the part that you think should be added to `onExternalSettingsChange`?

> [!note]
> For local development, the `id` should match the plugin's folder name; otherwise some methods, such as `onExternalSettingsChange`, won't be called.

**Comment by SashaKryzh** [comment]
> To confirm, this is roughly the part that you think should be added to `onExternalSettingsChange`?
> 
> Note
> 
> For local development, the `id` should match the plugin's folder name; otherwise some methods, such as `onExternalSettingsChange`, won't be called.

Yes. When it didn’t work for me, the first page I opened was the `onExternalSettingsChange` documentation.

Maybe some like this: _For local development, the `id` in the `manifest.json` should match the plugin's folder name; otherwise, this method won't be called._

**Comment by sigrunixia** [comment]
Got it, I'll add it to my batch of changes this evening as your PR was already merged. :)

**Comment by sigrunixia** [comment]
https://github.com/obsidianmd/obsidian-developer-docs/commit/478aaebead341f13bd4077e586fc022358d6b7f5
